### PR TITLE
Fix issue when ball is out

### PR DESCRIPTION
**Pull request by Ericfreespirit**
- Replace player when ball is out
- Put the ball at the init place in isCatch state
- Launch the ball when space is pressed

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -17,14 +18,16 @@ class FlameGame extends Game with KeyboardEvents, TapDetector {
...
+  bool isCatch = false;
+
   late double halfGameWidth;
 
   double direction = 0;
   double positionX = 500;
-  Vector2 ballPosition = Vector2(500, 500);
+  Vector2 ballPosition = Vector2(500, 600);
 
   double playerSpeed = 500;
```
Comment (mad221) [inline lib/main.dart:29]: I think we should to talk in percent of screen not in pixel directly think responsive 😉

File: lib/main.dart [hunk 2]
```diff
@@ -105,8 +112,19 @@ class FlameGame extends Game with KeyboardEvents, TapDetector {
     return distance <= circleRadius;
   }
 
+  bool ballIsOut(){
+      if (ballPosition.y < -300 || ballPosition.y > size.y * 1.5){
```
Comment (mad221) [inline lib/main.dart]: Same thingd here, speak screen position Is better and why under 300 pixels and not 0 ?

**Review by mad221 — COMMENTED** [review]
I think you can change the different think give in comment and that's OK ! Good job !

File: lib/main.dart (hunk 1 above)
Comment (Ericfreespirit) [reply lib/main.dart:29]: Ok why not Mister english

**Review by mad221 — APPROVED** [review]

**Pull request merged**